src/ray/common/scheduling/scheduling_ids.h

```diff
@@ -0,0 +1,50 @@
+#ifndef RAY_COMMON_SCHEDULING_SCHEDULING_IDS_H
+#define RAY_COMMON_SCHEDULING_SCHEDULING_IDS_H
+
+#include <iostream>
+#include <map>
+#include <unordered_map>
+#include <string>
+#include <iterator>
+#include <stdlib.h>
+#include <chrono>
+#include <vector>
+using namespace std;
+
+/// Class to map string IDs to unique integer IDs and back.
+class ScheduleIds {
```

Comment:
Consider calling this class `ResourceRegistry` or `ResourceIds`. That might be a better name. We should probably also have a name for the strings (I'd suggest `resource_name` or `resource_label`) and for the ids (I'd suggest `resource_id`).

Reply:
Actually, renamed it to StringIdMap 

Two reasons. First we use it not only for resource labels but also for client ids (so a little bit strange to call it resource_name). Second, it looks to me that this can be a general utility function we can use whenever we need to speedup code where objects are identified by strings.